### ISSUE TITLE
[JENKINS-43339] `currentBuild.result` is not set to `ABORTED` when the build was aborted

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/pipeline-model-definition-plugin#1044]
<p>Right now when the build is aborted using the `pipeline-input-step` plugin in a declarative pipeline, the build result is set to `FAILURE` instead of `ABORTED` in the post section. However, the actual result is properly set to `ABORTED` afterwards. This makes it impossible to do conditional code in the post section.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rochdev">rochdev</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339" target="_blank">`currentBuild.result` is not set to `ABORTED` when the build was aborted</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/abayer.png" title="abayer's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer">abayer</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: pipeline-input-step-plugin, pipeline-model-definition-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2017-12-18T17:53:56+00:00
<li><b>votes</b>: 6
<li><b>watchers</b>: 26
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Right now when the build is aborted using the `pipeline-input-step` plugin in a declarative pipeline, the build result is set to `FAILURE` instead of `ABORTED` in the post section. However, the actual result is properly set to `ABORTED` afterwards. This makes it impossible to do conditional code in the post section.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR #​153](https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/153)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rochdev:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=294302&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-294302">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=294302&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-294302">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It seems this is also true for `SUCCESS` which is set only after the build has completed (and after all post sections), but using `currentBuild.result` inside of a post section returns `null`.</p>
</pre>
</details>

</ul>
</details>

<p>It seems this is also true for `SUCCESS` which is set only after the build has completed (and after all post sections), but using `currentBuild.result` inside of a post section returns `null`.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>joc:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=296538&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296538">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=296538&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296538">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This also happens for builds that are Superseded (when using milestones).  When I get the current result status in the failure block it says "FAILURE" but the build final status is NOT_BUILT. (Same for ABORTED as above).</p>

<p>It's also incorrectly reporting as a completeBuild - which is an additional check that I'm performing.</p>

<p>Getting the previous build does return the correct result.</p>
</pre>
</details>

</ul>
</details>

<p>This also happens for builds that are Superseded (when using milestones).  When I get the current result status in the failure block it says "FAILURE" but the build final status is NOT_BUILT. (Same for ABORTED as above).</p>

<p>It's also incorrectly reporting as a completeBuild - which is an additional check that I'm performing.</p>

<p>Getting the previous build does return the correct result.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/abayer.png" title="abayer's avatar" /> abayer:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=296603&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296603">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=296603&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296603">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Interesting! We're setting the build result to <tt>FAILURE</tt> explicitly when we encounter an exception in execution of a stage, but I'm guessing that's not the ideal behavior for these scenarios. I'll investigate.</p>
</pre>
</details>

</ul>
</details>

<p>Interesting! We're setting the build result to <tt>FAILURE</tt> explicitly when we encounter an exception in execution of a stage, but I'm guessing that's not the ideal behavior for these scenarios. I'll investigate.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>joc:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=296604&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296604">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=296604&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296604">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thank you <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Thank you <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/abayer.png" title="abayer's avatar" /> abayer:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=296607&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296607">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=296607&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296607">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Tada - PR up at <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/153" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/153</a>.</p>
</pre>
</details>

</ul>
</details>

<p>Tada - PR up at <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/153" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/153</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=297001&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297001">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=297001&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297001">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Andrew Bayer
Path:
 pipeline-model-definition/pom.xml
 pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy
 pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/NotBuilt.groovy
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineFromDockerfileScript.groovy
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineScript.groovy
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/LabelScript.groovy
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/Messages.properties
 pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java
 pom.xml
<a href="http://jenkins-ci.org/commit/pipeline-model-definition-plugin/d9a8fb10d9c830470dfb9871696ca9df8c50f0f7" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/pipeline-model-definition-plugin/d9a8fb10d9c830470dfb9871696ca9df8c50f0f7</a>
Log:
  <span class="error">&#91;FIXED JENKINS-43339&#93;</span> Properly set result from FlowInterruptedException</p>

<p>We should be setting the build result to
FlowInterruptedException.getResult() when we catch a
FlowInterruptedException, since there's actually a relevant result
there. Also added a notBuilt post status while I was here.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Andrew Bayer<br/>
Path:<br/>
 pipeline-model-definition/pom.xml<br/>
 pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy<br/>
 pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/NotBuilt.groovy<br/>
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy<br/>
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineFromDockerfileScript.groovy<br/>
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineScript.groovy<br/>
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/LabelScript.groovy<br/>
 pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/Messages.properties<br/>
 pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/BasicModelDefTest.java<br/>
 pom.xml<br/>
<a href="http://jenkins-ci.org/commit/pipeline-model-definition-plugin/d9a8fb10d9c830470dfb9871696ca9df8c50f0f7" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/pipeline-model-definition-plugin/d9a8fb10d9c830470dfb9871696ca9df8c50f0f7</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-43339&#93;</span> Properly set result from FlowInterruptedException</p>

<p>We should be setting the build result to<br/>
FlowInterruptedException.getResult() when we catch a<br/>
FlowInterruptedException, since there's actually a relevant result<br/>
there. Also added a notBuilt post status while I was here.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/abayer.png" title="abayer's avatar" /> abayer:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=297923&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297923">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=297923&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297923">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Released in 1.1.4 just now!</p>
</pre>
</details>

</ul>
</details>

<p>Released in 1.1.4 just now!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>romanp:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=297968&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297968">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=297968&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297968">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I looked forward to this change, but my related problem is not fixed <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p><a href="http://stackoverflow.com/questions/42973501/declarative-pipeline-donts-send-post-failure-notificaitons-if-build-was-abbort" class="external-link" target="_blank" rel="nofollow noopener">http://stackoverflow.com/questions/42973501/declarative-pipeline-donts-send-post-failure-notificaitons-if-build-was-abbort</a></p>
</pre>
</details>

</ul>
</details>

<p>I looked forward to this change, but my related problem is not fixed <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p><a href="http://stackoverflow.com/questions/42973501/declarative-pipeline-donts-send-post-failure-notificaitons-if-build-was-abbort" class="external-link" target="_blank" rel="nofollow noopener">http://stackoverflow.com/questions/42973501/declarative-pipeline-donts-send-post-failure-notificaitons-if-build-was-abbort</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rbjorklin.png" title="rbjorklin's avatar" /> rbjorklin:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=297976&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297976">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=297976&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297976">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=romanp" class="user-hover" rel="romanp">romanp</a> I've been having the same problem as you and for me the issue you describe in stackoverflow was resolved with 1.1.4.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=romanp" class="user-hover" rel="romanp">romanp</a> I've been having the same problem as you and for me the issue you describe in stackoverflow was resolved with 1.1.4.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>romanp:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=297978&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297978">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=297978&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-297978">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>seems working again after restarting jenkins (maybe I've missed this after installing).
sorry for any inconvenience.</p>
</pre>
</details>

</ul>
</details>

<p>seems working again after restarting jenkins (maybe I've missed this after installing).<br/>
sorry for any inconvenience.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>akarola:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=315883&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315883">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=315883&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315883">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This issue still persist for us. We have script e.g.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">node {
    def repoGitUrl = <span class="code-quote">"git@usfr-git.foo.git"</span>

    jdk <span class="code-quote">"JDK8"</span>

    catchError {
        dir(<span class="code-quote">'FOO'</span>) {
            stage <span class="code-quote">"Checkout"</span>
            gitCheckout(repoGitUrl, <span class="code-quote">"master"</span>)

            stage <span class="code-quote">"Merge"</span>
            def gitflow = <span class="code-keyword">new</span> com.foo.GitFlow()
            gitflow.mergeMasterToOtherBranches()

            deleteDir()
        }

        currentBuild.result = <span class="code-quote">'SUCCESS'</span>
    }

    mailOnFailure(<span class="code-quote">"foo@bar.com"</span>)
}
</pre>
</div></div>
<div class="code panel" style="border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;"><b>The gitCheckout() function:</b></div><div class="codeContent panelContent">
<pre class="code-java">def call(gitlab, branch) {
    checkout([
            $class                           : <span class="code-quote">'GitSCM'</span>,
            branches                         : [[name: <span class="code-quote">"${branch}"</span>]],
            doGenerateSubmoduleConfigurations: <span class="code-keyword">false</span>,
            extensions                       : [
                    [$class: <span class="code-quote">'WipeWorkspace'</span>],
                    [$class: <span class="code-quote">'CleanBeforeCheckout'</span>],
                    [$class: <span class="code-quote">'LocalBranch'</span>, localBranch: <span class="code-quote">"${branch}"</span>]
            ],
            submoduleCfg                     : [],
            userRemoteConfigs                : [
                    [
                            credentialsId: <span class="code-quote">'GitLabBuildUser'</span>,
                            url          : <span class="code-quote">"${gitlab}"</span>
                    ]
            ]
    ])
}
</pre>
</div></div>
<div class="code panel" style="border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;"><b>The mailOnFailure() function:</b></div><div class="codeContent panelContent">
<pre class="code-java">def call(recipients) {
    <span class="code-keyword">if</span> (currentBuild.result != <span class="code-quote">'ABORTED'</span>) { <span class="code-comment">// don't send email when the build was aborted
</span>        step([$class                  : <span class="code-quote">'Mailer'</span>,
              notifyEveryUnstableBuild: <span class="code-keyword">true</span>,
              recipients              : recipients,
              sendToIndividuals       : <span class="code-keyword">true</span>])
    }
}
</pre>
</div></div>
<p> </p>

<p>When I abort the Jenkins job while it is checking out, the currentBuild.result is FAILURE, not ABORTED as expected. Git checkout throws a java.lang.InterruptedException and then the currentBuild.result contains FAILURE, not ABORTED.</p>

<p>Full log: <a href="https://pastebin.com/B5AFZnN9" class="external-link" target="_blank" rel="nofollow noopener">https://pastebin.com/B5AFZnN9</a></p>

<p> </p>
<div class="panel" style="border-width: 1px;"><div class="panelHeader" style="border-bottom-width: 1px;"><b>Versions:</b></div><div class="panelContent">
<p>pipeline-model-definition 1.1.9
 pipeline-input-step 2.7
 Jenkins 2.46.3</p>
</div></div>
</pre>
</details>

</ul>
</details>

<p>This issue still persist for us. We have script e.g.</p>

<pre>
node {
    def repoGitUrl = <span class="code-quote">"git@​usfr-git.foo.git"</span>

    jdk <span class="code-quote">"JDK8"</span>

    catchError {
dir(<span class="code-quote">'FOO'</span>) {
    stage <span class="code-quote">"Checkout"</span>
    gitCheckout(repoGitUrl, <span class="code-quote">"master"</span>)

    stage <span class="code-quote">"Merge"</span>
    def gitflow = <span class="code-keyword">new</span> com.foo.GitFlow()
    gitflow.mergeMasterToOtherBranches()

    deleteDir()
}

currentBuild.result = <span class="code-quote">'SUCCESS'</span>
    }

    mailOnFailure(<span class="code-quote">"foo@​bar.com"</span>)
}
</pre>
<div class="code panel" style="border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;"><b>The gitCheckout() function:</b></div><div class="codeContent panelContent">
<pre class="code-java">def call(gitlab, branch) {
    checkout([
    $class   : <span class="code-quote">'GitSCM'</span>,
    branches : [[name: <span class="code-quote">"${branch}"</span>]],
    doGenerateSubmoduleConfigurations: <span class="code-keyword">false</span>,
    extensions       : [
    [$class: <span class="code-quote">'WipeWorkspace'</span>],
    [$class: <span class="code-quote">'CleanBeforeCheckout'</span>],
    [$class: <span class="code-quote">'LocalBranch'</span>, localBranch: <span class="code-quote">"${branch}"</span>]
    ],
    submoduleCfg     : [],
    userRemoteConfigs: [
    [
    credentialsId: <span class="code-quote">'GitLabBuildUser'</span>,
    url  : <span class="code-quote">"${gitlab}"</span>
    ]
    ]
    ])
}
</pre>
</div></div>
<div class="code panel" style="border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;"><b>The mailOnFailure() function:</b></div><div class="codeContent panelContent">
<pre class="code-java">def call(recipients) {
    <span class="code-keyword">if</span> (currentBuild.result != <span class="code-quote">'ABORTED'</span>) { <span class="code-comment">// don't send email when the build was aborted
</span>step([$class  : <span class="code-quote">'Mailer'</span>,
      notifyEveryUnstableBuild: <span class="code-keyword">true</span>,
      recipients      : recipients,
      sendToIndividuals       : <span class="code-keyword">true</span>])
    }
}
</pre>
</div></div>
<p> </p>

<p>When I abort the Jenkins job while it is checking out, the currentBuild.result is FAILURE, not ABORTED as expected. Git checkout throws a java.lang.InterruptedException and then the currentBuild.result contains FAILURE, not ABORTED.</p>

<p>Full log: <a href="https://pastebin.com/B5AFZnN9" class="external-link" target="_blank" rel="nofollow noopener">https://pastebin.com/B5AFZnN9</a></p>

<p> </p>


<table><tr><td><b>Versions:</b></td></tr><tr><td><p>pipeline-model-definition 1.1.9<br/>
 pipeline-input-step 2.7<br/>
 Jenkins 2.46.3</p></td></tr></table>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pjohnston:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=318844&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-318844">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=318844&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-318844">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This happens for me right now as well with Jenkins 2.72.3. </p>
</pre>
</details>

</ul>
</details>

<p>This happens for me right now as well with Jenkins 2.72.3. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pwilcock:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=322491&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-322491">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=322491&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-322491">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Also confirming this error with declarative pipelines. When I abort the build the post-build action for failure triggers, not abort.  </p>

<p>Final two lines of console log show:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">ERROR: script returned exit code 143
Finished: ABORTED
</pre>
</div></div>
<p>Jenkins 2.93. 
Latest pipeline plugins all installed as of this date. </p>
</pre>
</details>

</ul>
</details>

<p>Also confirming this error with declarative pipelines. When I abort the build the post-build action for failure triggers, not abort.  </p>

<p>Final two lines of console log show:</p>

<pre>
ERROR: script returned exit code 143
Finished: ABORTED
</pre>
<p>Jenkins 2.93. <br/>
Latest pipeline plugins all installed as of this date. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/abayer.png" title="abayer's avatar" /> abayer:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=323115&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323115">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=323115&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323115">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Regrettably, not much we can do here - the <tt>ABORTED</tt> status, in some cases, doesn't get set until the Pipeline finishes executing, so we can't tell from Declarative that we should be treating the build as <tt>ABORTED</tt>.</p>
</pre>
</details>

</ul>
</details>

<p>Regrettably, not much we can do here - the <tt>ABORTED</tt> status, in some cases, doesn't get set until the Pipeline finishes executing, so we can't tell from Declarative that we should be treating the build as <tt>ABORTED</tt>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pjohnston:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=323121&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323121">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=323121&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323121">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer" class="user-hover" rel="abayer">abayer</a>, rather than just saying this won't be fixed, perhaps it is better to remove features that provide the illusion that `ABORTED` is a supported state that can be handled in declarative pipelines? Then there will be no confusion <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer" class="user-hover" rel="abayer">abayer</a>, rather than just saying this won't be fixed, perhaps it is better to remove features that provide the illusion that `ABORTED` is a supported state that can be handled in declarative pipelines? Then there will be no confusion <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pjohnston:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=323122&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323122">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=323122&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323122">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>(I suspect this issue will continue to be reported when someone else tries to handle `aborted` in a `post` block)</p>
</pre>
</details>

</ul>
</details>

<p>(I suspect this issue will continue to be reported when someone else tries to handle `aborted` in a `post` block)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/abayer.png" title="abayer's avatar" /> abayer:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=323126&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323126">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=323126&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323126">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The thing is that <tt>ABORTED</tt> <b>does</b> work in some contexts, but not others. It's weird.</p>
</pre>
</details>

</ul>
</details>

<p>The thing is that <tt>ABORTED</tt> <b>does</b> work in some contexts, but not others. It's weird.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pwilcock:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=323127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323127">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=323127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-323127">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer" class="user-hover" rel="abayer">abayer</a> I've observed the following:</p>

<p>When running parallel steps with fastFail set, the fast-failed aborted build will return FAILURE if the act of aborting a step that was <em>still in progress</em> returns an error code. </p>

<p>However if other parallel steps have finished successfully, and the last running step fails, this correctly returns ABORTED in the event of failure.</p>

<p>Some kind of race condition where if a step fails, fastFail aborts all other steps, and the last error code returned is the status seen by the declarative pipeline, but isn't the one respected by the overall build.status? Hopefully that makes sense. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer" class="user-hover" rel="abayer">abayer</a> I've observed the following:</p>

<p>When running parallel steps with fastFail set, the fast-failed aborted build will return FAILURE if the act of aborting a step that was <em>still in progress</em> returns an error code. </p>

<p>However if other parallel steps have finished successfully, and the last running step fails, this correctly returns ABORTED in the event of failure.</p>

<p>Some kind of race condition where if a step fails, fastFail aborts all other steps, and the last error code returned is the status seen by the declarative pipeline, but isn't the one respected by the overall build.status? Hopefully that makes sense. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=344373&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-344373">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=344373&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-344373">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is there not a reliable way to get from the rawBuild if a build was aborted?
I cannot image that there's <b>no</b> way - as the builds are eventually - when they are totally finished - always displayed with the right result. 'Only' currentBuild.currentResult is very often giving a different result (FAILURE usually)
We're using this hack now to get better detection of aborted builds:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">if</span> (currentBuild.rawBuild.getActions(jenkins.model.InterruptedBuildAction.class).isEmpty()) {
   <span class="code-keyword">return</span> currentBuild.currentResult
} <span class="code-keyword">else</span> {
  <span class="code-keyword">return</span> <span class="code-quote">"ABORTED"</span>
}
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Is there not a reliable way to get from the rawBuild if a build was aborted?<br/>
I cannot image that there's <b>no</b> way - as the builds are eventually - when they are totally finished - always displayed with the right result. 'Only' currentBuild.currentResult is very often giving a different result (FAILURE usually)<br/>
We're using this hack now to get better detection of aborted builds:</p>


<pre>
<span class="code-keyword">if</span> (currentBuild.rawBuild.getActions(jenkins.model.InterruptedBuildAction.class).isEmpty()) {
   <span class="code-keyword">return</span> currentBuild.currentResult
} <span class="code-keyword">else</span> {
  <span class="code-keyword">return</span> <span class="code-quote">"ABORTED"</span>
}
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/bitwiseman.png" title="bitwiseman's avatar" /> bitwiseman:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=378462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-378462">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=378462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-378462">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Bulk closing resolved issues. </p>
</pre>
</details>

</ul>
</details>

<p>Bulk closing resolved issues. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>reinholdfuereder:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=414395&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-414395">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=414395&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-414395">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi" class="user-hover" rel="kutzi">kutzi</a> – this workaround is indeed very helpful! (Of course one would prefer a proper/clean solution/fix... <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/>)</p>
</pre>
</details>

</ul>
</details>

<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi" class="user-hover" rel="kutzi">kutzi</a> – this workaround is indeed very helpful! (Of course one would prefer a proper/clean solution/fix... <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/>)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=415056&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-415056">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=415056&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-415056">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=reinholdfuereder" class="user-hover" rel="reinholdfuereder">reinholdfuereder</a> according to your comment, you still have experienced this issue in the current release?
 I'm having the workaround in place and therefore not watching it anymore.
 According to Liam's comment this should have been fixed?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=reinholdfuereder" class="user-hover" rel="reinholdfuereder">reinholdfuereder</a> according to your comment, you still have experienced this issue in the current release?<br/>
 I'm having the workaround in place and therefore not watching it anymore.<br/>
 According to Liam's comment this should have been fixed?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/aspiridonovs.jpg" title="aspiridonovs's avatar" /> aspiridonovs:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/43339?focusedId=434475&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-434475">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-43339?focusedId=434475&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-434475">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Jenkins 2.375.1, manually aborted build had `currentBuild.currentResult`= success, sending notification as such. Workaround helped. </p>
</pre>
</details>

</ul>
</details>

<p>Jenkins 2.375.1, manually aborted build had `currentBuild.currentResult`= success, sending notification as such. Workaround helped. </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-46572%22">JENKINS-46572</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-43895%22">JENKINS-43895</a>]</i>
